verify/assets/index.html

```diff
@@ -116,153 +155,161 @@ <h2>Twilio Verify</h2>
     const phoneInput = window.intlTelInput(phoneInputField, {
       // https://en.wikipedia.org/wiki/ISO_3166-1_alpha-2
       preferredCountries: ["us", "co", "in", "de"],
-      separateDialCode: true,
       utilsScript:
-        "https://cdnjs.cloudflare.com/ajax/libs/intl-tel-input/16.0.11/js/utils.js"
+        "https://cdn.jsdelivr.net/npm/intl-tel-input@17.0.13/build/js/utils.js",
     });
 
-    $("#channel-select").change(function() {
-      const channel = $("input[name='channel']:checked").val();
-      if (channel === "email") {
-        $(".phone-input").hide();
-        $(".email-input").show();
-      } else if (channel === "sms" || channel === "call") {
-        $(".phone-input").show();
-        $(".email-input").hide();
-      }
-    });
+    function selectedChannel() {
+      const checked = "input[name='channel']:checked";
+      return document.querySelector(checked).value;
+    }
+
+    function showEmailInput() {
+      document.getElementById("phone-input").style.display = "none";
+      document.getElementById("email-input").style.display = "block";
+    }
+
+    function showPhoneNumberInput() {
+      document.getElementById("phone-input").style.display = "block";
+      document.getElementById("email-input").style.display = "none";
+    }
+
+    function showModalStatus(message, color = "gray") {
+      let status = document.getElementById("modal-status");
+      status.style.color = color;
+      status.textContent = message;
+    }
+
+    function showError(error) {
+      console.error(error);
+      showStatus(error, (color = "#a94442"));
```

Comment:
You're actually creating a global `color` variable at this point, is that for any reason?

It looks to me as though you are doing this to self document what is going on (you're passing a color to the `showStatus` function). If you want to be more explicit with this, I'd probably name the actual color too, so you don't have to parse the hex yourself. But I'd also do it with a local variable. Maybe like this:

```js
function showError(error) {
  console.error(error)
  const redColor = "#a94442";
  showStatus(error, redColor);
}
```

An alternative, if you just want to make it more obvious that you are passing a color would be to define `showStatus` with an object of options:

```js
function showStatus(message, options = { color: "gray" }) {
  let status = document.getElementById("status");
  status.style.color = options.color;
  status.textContent = message;
};
```

Then you can call `showStatus` with `showStatus(error, { color: "#a98882" });`.

Reply:
thanks for this detailed explanation, this is entirely my misunderstanding of how javascript function arguments work but now I know!